File: app/src/main/java/com/example/user/eventest/widget/WidgetData.java
package com.example.user.eventest.widget;

import com.example.user.eventest.model.Memo;
import com.example.user.eventest.model.RoomRepository;

import java.util.ArrayList;

/**
 * Created by DR
 * on 28.02.2018.
 */

public class WidgetData implements WidgetPresenter {

    private RoomRepository roomRepository;

    WidgetData(RoomRepository roomRepository) {
        this.roomRepository = roomRepository;
    }

    @Override
    public ArrayList<Memo> populateListItem() {

        return roomRepository.getAllData();

    }
}

File: app/src/main/java/com/example/user/eventest/widget/WidgetData.kt
package com.example.user.eventest.widget

import com.example.user.eventest.model.Memo
import com.example.user.eventest.model.RoomRepository
import java.util.*

/**
 * Created by DR
 * on 28.02.2018.
 */

class WidgetData(private val roomRepository: RoomRepository) : WidgetPresenter {

    override fun populateListItem(): ArrayList<Memo> = roomRepository.getAllData()

}
